framework/DataObjects/DataSet.py

```diff
@@ -1559,26 +1575,30 @@ def _getRealizationFromCollectorByIndex(self,index):
       rlz[var] = vals
     return rlz
 
-  def _getRealizationFromCollectorByValue(self, toMatch, noMatch, tol=1e-15):
+  def _getRealizationFromCollectorByValue(self, toMatch, noMatch, tol=1e-15, options=None):
     """
       Obtains a realization from the collector storage matching the provided index
       @ In, toMatch, dict, elements to match
       @ In, noMatch, dict, elements to AVOID matching (should not match within tolerance)
       @ In, tol, float, optional, tolerance to which match should be made
+      @ In, options, dict, optional, options to be applied to the search
       @ Out, r, int, index where match was found OR size of data if not found
       @ Out, rlz, dict, realization as {var:value} OR None if not found
     """
     if toMatch is None:
       toMatch = {}
     if noMatch is None:
       noMatch = {}
+      allMatch = True
```

Comment:
Hm, should this be:
```
    if noMatch == {}
      allMatch = True
    if noMatch is None:
      noMatch = {}
      ```
instead of:
```
    if noMatch is None:
      noMatch = {}
      allMatch = True
      ```

Reply:
the else got added while I was reading this.

Reply:
For some reason I thought I had this:
def _getRealizationFromCollectorByValue(self, toMatch, noMatch=None, tol=1e-15, options=None):
instaed of this
def _getRealizationFromCollectorByValue(self, toMatch, noMatch, tol=1e-15, options=None):

Reply:
Added, thank you josh